File: Codeforces/Round #636 (Div. 3)/C.py
def sign(x):
    return (x>0)

t = int(input())

while(t):
    n = int(input())
    
    #leitura do array em uma linha
    arr = [int(x) for x in input().split()]

    sum=0
    maiorAtual = arr[0]
    ultimo = arr[0]

    for i in arr[1:]:
        if(sign(i)==sign(ultimo)):
            maiorAtual = max(maiorAtual,i)
        else:
            sum+=maiorAtual
            maiorAtual = i
        ultimo = i

    sum+=maiorAtual
    print(sum)
    t-=1

File: Codeforces/Round #636 (Div. 3)/C.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;

bool sign(ll x){
    return ((x>0) ? true : false);
}

int main(){
    ll t, n;

    cin >> t;
    while(t--){
        cin >> n;
        ll sum=0, current, last, x;

        cin >> last;
        current = last;
        for(int i=0;i<n-1;i++){
            cin >> x;
            if(sign(x)==sign(last)){
                current = max(current,x);
            }
            else{
                sum+=current;
                current = x;
            }
            last = x;
        }

        cout << sum + current <<'\n';
    }

    return 0;
}